hrms/payroll/doctype/payroll_entry/payroll_entry.py

```diff
@@ -1158,6 +1158,7 @@ def get_filtered_employees(
 	query = set_fields_to_select(query, fields)
 	query = set_searchfield(query, searchfield, search_string, qb_object=Employee)
 	query = set_filter_conditions(query, filters, qb_object=Employee)
+	salary_withholding_documents = frappe.db.get_all("Salary Withholding")
```

Comment:
The date filtering for fetching salary withholding documents should happen here itself instead of fetching all records

Reply:
I meant the date filtering should happen in this `get_all` query based on dates instead of looping and filtering in python later. 

Not "join" in the main query since this won't be a widely used feature initially